capsules/src/lsm303agr.rs

```diff
@@ -216,8 +219,9 @@ impl<'a> Lsm303agrI2C<'a> {
             // turn on i2c to send commands
             buf[0] = 0x0F;
             self.i2c_magnetometer.enable();
-            // TODO verify errors
-            let _ = self.i2c_magnetometer.write_read(buf, 1, 1);
+            if let Err((error, buf)) = self.i2c_magnetometer.write_read(buf, 1, 1) {
+                self.command_complete(buf, Err(error));
```

Comment:
Calling a callback in a downcall is a bug in Tock, this error would need to be delivered via a deferred call (see https://github.com/tock/tock/blob/master/doc/reference/trd-hil-design.md#rule-1-dont-make-synchronous-callbacks). Same goes for the other `command_complete()` calls in this file that can happen immediately in response to a downcall

Reply:
Done